pyminio/main.py

```diff
@@ -307,25 +306,34 @@ def rmdir(self, path: str, recursive: bool = False):
             raise DirectoryNotEmptyError("can not recursively delete "
                                          "unempty directory")
 
-        objects_in_directory = self._get_objects_at(match)
+        dirs_to_delete = [path]
 
-        if len(objects_in_directory) > 0:
-            if not recursive:
-                raise DirectoryNotEmptyError("can not recursively delete "
-                                             "unempty directory")
+        while len(dirs_to_delete):
+            current_dir_match = Match(dirs_to_delete.pop(0))
+            objects_in_directory = self._get_objects_at(current_dir_match)
+            files = []
+            dirs = []
 
-            files = [file_obj.object_name
-                     for file_obj in objects_in_directory
-                     if not file_obj.is_dir]
+            if len(objects_in_directory) > 0:
+                if not recursive:
+                    raise DirectoryNotEmptyError("can not recursively delete "
```

Comment:
The error message isn't clear
find another word for unempty

Reply:
non-empty